app/preprints/-components/submit/metadata/component.ts

```diff
@@ -143,9 +143,12 @@ export default class Metadata extends Component<MetadataArgs>{
         }
     }
 
-    @action
-    hasSubjects(): void {
-        this.validate();
+    @task
+    @waitFor
+    public async hasSubjects(hasSubjects: boolean): Promise<void> {
```

Comment:
I think this can stay as a non-async `@action`, as there isn't any async activity here